dbt/adapters/fabric/fabric_connection_manager.py

```diff
@@ -30,6 +30,27 @@
 
 logger = AdapterLogger("fabric")
 
+# https://github.com/mkleehammer/pyodbc/wiki/Data-Types
+datatypes = {
+    # "str": "char",
+    "str": "varchar",
+    "uuid.UUID": "uniqueidentifier",
+    "uuid": "uniqueidentifier",
+    # "float": "real",
+    # "float": "float",
+    "float": "bigint",
+    # "int": "smallint",
+    # "int": "tinyint",
+    "int": "int",
+    "bytes": "varbinary",
+    "bool": "bit",
+    "datetime.date": "date",
+    "datetime.datetime": "datetime2(6)",
+    "datetime.time": "time",
+    "decimal.Decimal": "decimal",
+    # "decimal.Decimal": "numeric",
```

Comment:
I'm assuming the commenting is from your testing? that said, if there's any knowledge in your brain about why specific types are used opposed to others, here would be a great place to add that info as a comment for... posterity

```suggestion
    "str": "varchar",
    "uuid.UUID": "uniqueidentifier",
    "uuid": "uniqueidentifier",
    "float": "bigint",
    "int": "int",
    "bytes": "varbinary",
    "bool": "bit",
    "datetime.date": "date",
    "datetime.datetime": "datetime2(6)",
    "datetime.time": "time",
    "decimal.Decimal": "decimal",
```

Reply:
Although the same key can have multiple values, the data_type_code_to_name picks the first value for asserting model constraints. I have aligned the tests to match uncommented data types but want to keep others for future reference.